### ISSUE TITLE
[BUG] SnapshotStatusApisIT.testExceptionOnMissingShardLevelSnapBlob fails with reproducible seed

### ISSUE THREAD

**Issue by mch2** [opensearch-project/OpenSearch#15584]
### Describe the bug

```
junit.framework.AssertionFailedError: Expected exception SnapshotMissingException but no exception was thrown
	at __randomizedtesting.SeedInfo.seed([5F17849827BA79F3:C1C9BD258865E37A]:0)
	at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2886)
	at org.apache.lucene.tests.util.LuceneTestCase.expectThrows(LuceneTestCase.java:2872)
	at org.opensearch.snapshots.SnapshotStatusApisIT.testExceptionOnMissingShardLevelSnapBlob(SnapshotStatusApisIT.java:211)
	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
```

```
REPRODUCE WITH: ./gradlew ':server:internalClusterTest' --tests "org.opensearch.snapshots.SnapshotStatusApisIT.testExceptionOnMissingShardLevelSnapBlob" -Dtests.seed=5F17849827BA79F3 -Dtests.security.manager=true -Dtests.jvm.argline="-XX:TieredStopAtLevel=1 -XX:ReservedCodeCacheSize=64m" -Dtests.locale=it-VA -Dtests.timezone=US/Hawaii -Druntime.java=21
```

### Related component

Storage:Snapshots

### To Reproduce

Seed above.

### Expected behavior

Test to pass

**Comment by prudhvigodithi** [comment]
Hey @mch2 the automation flagged the class `SnapshotStatusApisIT` showing some flaky tests, please check https://github.com/opensearch-project/OpenSearch/issues/15815. Can we close this issue?
Thanks
@dblock @rednaksi91 @andrross @getsaurabh02

**Comment by mch2** [comment]
@prudhvigodithi this fails with a reproducible seed so imo should be treated as a bug not a flaky test.

**Comment by prudhvigodithi** [comment]
Coming from here https://github.com/opensearch-project/OpenSearch/issues/15815, the `SnapshotStatusApisIT` failed at the post merge action which means it passed at one point, the PR was merged and started to see some failures after then. The issue also has the other PR's linked on which the `SnapshotStatusApisIT` failed at one point and passed after re-try as some of them are in merged state.
Thanks

**Comment by ashking94** [comment]
@ltaragi Lets close this issue if it has been fixed. 
[Triage - attendees [1](https://github.com/ashking94) [2](https://github.com/linuxpi) [3](https://github.com/ashking94) [4](https://github.com/sachinpkale) [5](https://github.com/skumawat2025)]

**Comment by ltaragi** [comment]
Auto-cut issue (#15815) was closed by PR (#16146). Closing this one as well.
